smartmon.sh

```diff
@@ -69,6 +69,20 @@ SMARTMONATTRS
 )"
 smartmon_attrs="$(echo "${smartmon_attrs}" | xargs | tr ' ' '|')"
 
+# We should not hardcode the smartctl binary path,
+# instead for those OS that does not follow the
+# same default installation path, we should add it here.
+os_detect() {
+        if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+                smartctl="/usr/sbin/smartctl"
+        elif [[ "$OSTYPE" == "freebsd"* ]]; then
+                smartctl="/usr/local/sbin/smartctl"
+        else
+                smartctl="/usr/sbin/smartctl"
+        fi
+}
```

Comment:
Why not simply omit the absolute path from the `smartctl` invocations, and let the OS shell PATH find it?